src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
```

Comment:
PullProofAI

The implementation of the `FeedbackSecretGuard` is straightforward and correctly checks the `x-feedback-secret` header against the expected secret value. However, there are a few improvements that can be suggested:

1. **Error Handling Improvement**: Instead of throwing an exception directly, consider returning a `false` value when the secret is invalid. This aligns with the purpose of the `CanActivate` interface, which expects a boolean return value.

2. **Consistency Enhancement**: Ensure consistent error messaging and handling across guards. Consider using a centralized error handling mechanism to maintain uniformity in error responses.

3. **Security Enhancement**: Depending on the sensitivity of the operation protected by this guard, you may want to log unauthorized access attempts for auditing purposes.

By incorporating these suggestions, you can enhance the guard's maintainability, consistency, and security posture.


<!-- This is an auto-generated comment by PullProofAI -->